### ISSUE TITLE
rustup stable components not available

### ISSUE THREAD

**Issue by cyplo** [NixOS/nixpkgs#59005]
## Issue description
There was a component stabilization effort in `rustup` some time ago the components with `-preview` suffix were stabilized and do not need this suffix anymore. This was a part of getting Rust 2018 edition out, making sure that the basic tools used by the community are polished.

It seems that the stable, a.k.a non-preview components are not seen by rustup on NixOS using this (nixpkgs) repo. 

Components I found where the stable version is not available: `rls`, `clippy`, `rustfmt`

Not sure if rustup itself needs updating or what the cause is tbh - but happy to help debugging if needed.

### Steps to reproduce
* install rustup from this (nixpkgs) repo (I use the tarball approach to use the repo for just some packages)

```bash
$ rustp --version
rustup 1.17.0 ( )
$ rustfmt
error: 'rustfmt' is not installed for the toolchain 'stable-x86_64-unknown-linux-gnu'
To install, run `rustup component add rustfmt`
$ rustup component add rustfmt
error: toolchain 'stable-x86_64-unknown-linux-gnu' does not contain component 'rustfmt' for target 'x86_64-unknown-linux-gnu'
```

## Technical details
```
 - system: `"x86_64-linux"`
 - host os: `Linux 4.14.109, NixOS, 18.09.2473.58efb6980fb (Jellyfish)`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.1.3`
 - channels(cyryl): `""`
 - channels(root): `"nixos-18.09.2473.58efb6980fb"`
 - nixpkgs: `/nix/var/nix/profiles/per-user/root/channels/nixos`
```

**Comment by symphorien** [comment]
rustup is in nixpkgs. This version works:
```
$  rm -r ~/.rustup
$  rustup default stable-x86_64-unknown-linux-gnu
info: syncing channel updates for 'stable-x86_64-unknown-linux-gnu'
info: latest update on 2019-02-28, rust version 1.33.0 (2aa4c46cf 2019-02-28)
info: downloading component 'rustc'
 84.7 MiB /  84.7 MiB (100 %)   1.4 MiB/s ETA:   0 s                
info: downloading component 'rust-std'
 56.8 MiB /  56.8 MiB (100 %)   1.4 MiB/s ETA:   0 s                
info: downloading component 'cargo'
  4.4 MiB /   4.4 MiB (100 %)   1.4 MiB/s ETA:   0 s                
info: downloading component 'rust-docs'
  8.5 MiB /   8.5 MiB (100 %)   1.3 MiB/s ETA:   0 s                
info: installing component 'rustc'
info: installing component 'rust-std'
info: installing component 'cargo'
info: installing component 'rust-docs'
info: default toolchain set to 'stable-x86_64-unknown-linux-gnu'

  stable-x86_64-unknown-linux-gnu installed - rustc 1.33.0 (2aa4c46cf 2019-02-28)

$  rustup component add rustfmt                  
info: downloading component 'rustfmt'
  2.2 MiB /   2.2 MiB (100 %)   1.5 MiB/s ETA:   0 s                
info: installing component 'rustfmt'
$  rustfmt --help
Format Rust code

usage: /home/symphorien/.rustup/toolchains/stable-x86_64-unknown-linux-gnu/bin/rustfmt [options] <file>...
```
If you use nixpkgs's rustc and cargo, rustfmt and rls are available in 19.03. clippy is likely to never be in nixpkgs because it is too dependent on nightly features, and nixpkgs does not have nightly rust. You might also be interested in mozilla's rust overlay.

**Comment by cyplo** [comment]
It may indeed have been an older version of rustup lingering on my system. I did `rm -r ~/.rustup` and it helped, without reinstalling rustup. Thank you !
